controller_manager:
  ros__parameters:
    update_rate: 100

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    forward_position_controller:
      type: forward_position_controller/JointTrajectoryController

forward_position_controller:
  ros__parameters:
    joints:
      - slewing_joint
      - trolley_joint
      - hook_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    allow_partial_joints_goal: false
    constraints:
      goal_time: 2.0
      stopped_velocity_tolerance: 0.01
      slewing_joint:
        trajectory: 0.1
        goal: 0.05
      trolley_joint:
        trajectory: 0.02
        goal: 0.01
      hook_joint:
        trajectory: 0.02
        goal: 0.01
